projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/pojo/CreateEnvironmentRequest.java

```diff
@@ -0,0 +1,110 @@
+package org.batfish.datamodel.pojo;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.common.base.MoreObjects;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.batfish.datamodel.Edge;
+import org.batfish.datamodel.collections.NodeInterfacePair;
+
+/**
+ * The {@link CreateEnvironmentRequest CreateEnvironmentRequest} is an Object representation of the
+ * request used to create an environment and update an existing environment for BatFish service.
+ */
+public class CreateEnvironmentRequest {
+
+  private final String _name;
+  private final List<Edge> _edgeBlacklist;
+  private final List<NodeInterfacePair> _interfaceBlacklist;
+  private final List<String> _nodeBlacklist;
+  private final List<TableObject> _bgpTables;
+  private final List<TableObject> _routingTables;
+  private final String _externalBgpAnnouncements;
+
+  @JsonCreator
+  public CreateEnvironmentRequest(
```

Comment:
Since we don't have any setters and no zero arguments constructor, how will we create an Environmentrequest object if we need to create it incrementally ? This will be required as in service we are reading from the filestream and incrementally populating various Environment attributes